### PR TITLE
Add GitHub Username feature

### PR DESCRIPTION
**Pull request by Teee728**
Introduce a new feature to display GitHub usernames for each person in the address book. The feature ensures that usernames are limited to alphanumeric characters and dashes, in alignment with GitHub's username requirements.

Unit tests have been added to validate the functionality of the new feature.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -1,11 +1,7 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:4]: Change import statement

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -1,11 +1,7 @@
 package seedu.address.logic.parser;
 
 import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:4]: this too!

**Review by maiyasaliha — CHANGES_REQUESTED** [review]
Just modify the import statements, but otherwise LGTM!

**Review by yuechen2001 — DISMISSED** [review]

**Review by yuechen2001 — APPROVED** [review]

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 3]
```diff
@@ -36,11 +38,13 @@ class JsonAdaptedPerson {
     @JsonCreator
     public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
             @JsonProperty("email") String email, @JsonProperty("address") String address,
+            @JsonProperty ("github username") String gitHubUsername,
```
Comment (angsongyee) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:41]: Maybe github_username would be more suitable for the json key
```suggestion
            @JsonProperty ("github-username") String gitHubUsername,
```

**Review by angsongyee — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java (hunk 3 above)
Comment (angsongyee) [reply src/main/java/seedu/address/storage/JsonAdaptedPerson.java:41]: Also including all the other occurences of `"github username"`.

**Pull request merged**